src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -1061,7 +1084,15 @@ public synchronized void handle(Message<T> event) {
         }
       } else {
         checkNextTick();
-        handler.handle(event);
+        MessageImpl abc = (MessageImpl) event;
+        metrics.beginHandleMessage(metric, abc.getSocket() == null);
+        try {
+          handler.handle(event);
+          metrics.endHandleMessage(metric, null);
```

Comment:
shouldn't the processing time of the callback be effectively zero (or very small), unless the user is blocking the event loop?

And, if they are blocking the event loop, the event loop blocking code will kick in anyway and warn them.